server/src/main/java/org/elasticsearch/health/metadata/HealthMetadataService.java

```diff
@@ -53,25 +54,31 @@ public class HealthMetadataService {
     private volatile boolean enabled;
 
     // Signifies that a node has been elected as master, but it was not able yet to publish its health metadata for
-    // other reasons for example not all nodes of the cluster are 8.4.0 or newer
-    private volatile boolean readyToPublish = false;
+    // other reasons for example not all nodes of the cluster are 8.5.0 or newer
+    private volatile boolean readyToPublishAsMaster = false;
```

Comment:
Looking at where this is used, I think we can get rid of this flag (I'll suggest below how `updateOnClusterStateChange` could look like so it's not used anymore)